pkg/daemon/ceph/client/pool.go

```diff
@@ -183,11 +183,24 @@ func CreatePool(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec
 }
 
 func CreatePoolWithPGs(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec, appName, pgCount string) error {
+	autoscaleMode := pool.AutoscaleMode
 	if pool.Name == "" {
 		return errors.New("pool name must be specified")
 	}
+	if pool.Parameters != nil {
+		placementGroupCount, ok := pool.Parameters["pg_num"]
```

Comment:
I'm still really hesitant to set the auto-scale mode based on pg_num without understanding exactly what Ceph's built-in behavior is. I want to look at the Ceph code and see if they do this logic already. It's far more preferred to allow ceph to manage itself than force its hand.

Reply:
There are some potential negative effects to be aware of when using Ceph PG autoscaling:

Increased latency: Autoscaling can result in increased latency if it takes time for Ceph to rebalance the data across the cluster.
Increased network traffic: Autoscaling can result in increased network traffic as data is rebalanced across the cluster, which can impact the performance of other applications on the same network.
Increased storage overhead: Autoscaling can result in increased storage overhead, as additional placement groups are created to store data, which can lead to increased storage costs.
Complexity: Autoscaling can add complexity to Ceph's architecture, as it requires the use of additional tools and services to monitor and manage the placement groups.
Management: Autoscaling can be difficult to manage in Ceph, especially if there are multiple clusters with different usage patterns.

so we wan to conrtol the pg_num by user. @BlaineEXE

Reply:
Is this from a Ceph doc? If so, it would be good to link to the Ceph doc that mentions this in the Rook docs.

Reply:
@BlaineEXE Experience gained from practical tests, the pg_num growth too slow， when the pg_num growth, the data rebalance will effect r/w performance.

Reply:
Thanks for that info. I'll see if that can easily be suggested to the Ceph project to include those details.